### ISSUE TITLE
Allow hosting addons on LAN without HTTPS

### ISSUE THREAD

**Issue by nyakaspeter** [Stremio/stremio-features#687]
**Stremio Version**
All versions

**Is your feature request related to a problem? Please describe.**
I developed an addon for Stremio because the built-in torrent client does not play well with private trackers (I know this is because private trackers don't support magnet links). [My addon ](https://github.com/nyakaspeter/stremio-torrent-stream) connects to a locally hosted [HTTP API](https://github.com/nyakaspeter/torrent-stream-server) that uses WebTorrent to create media streams from torrent files on the fly. This works great however I can only install the addon in Stremio if it is running on localhost or if it's served over HTTPS. I don't want to expose the addon nor the streaming API to the internet I just want to self-host it on my LAN.

**Describe the solution you'd like**
Allow the installation of addons that are running on the local network, even if they are served over plain HTTP. Or at least make this optional.

**Describe alternatives you've considered**
I tried putting the addon server behind a reverse proxy to host it over HTTPS. This way it works fine on my PC, my phone and on my TV too. But it's not ideal as it requires owning a domain name and exposing the addon to the internet, or fiddling with the DNS settings of every client device.

**Comment by jaruba** [comment]
I'm afraid that this is not possible. The first issue is that Stremio is a security and privacy project, so this is a choice we made a long time ago for the purpose of ensuring the user's security. But this is not the only issue, some Stremio projects (including but not limited to Stremio Web) simply cannot interact with addons that do not have a valid (not self-signed) SSL certificate due to mixed content errors. A website using https cannot retrieve resources from an insecure http server, the browser itself will block this request, browsers only allow connections without SSL from a HTTPS website if the responses are coming from 127.0.0.1 or localhost, just like Stremio does..

**Comment by nyakaspeter** [comment]
Okay I understand web based applications require proper HTTPS connections to work. But the native applications on Windows and Android technically have no problem with connecting to HTTP only servers right? If all traffic is happening over LAN security shouldn't be an issue unless of course if the local network is compromised. Maybe this could be an advanced feature on the platforms that support this under a 'developer mode' toggle?

In the meanwhile I found [localtunnel](https://github.com/localtunnel/localtunnel) which simplifies serving the addon over HTTPS, but it'd be great not having to rely on a third-party service.

**Comment by jaruba** [comment]
localtunel and ngrok can serve as free reverse proxies, although we wouldn't recommend it due to security concerns
i would say that the solution would rather be to have an SSL certificate for the addon's LAN IP, i remember a fun project from some years ago called PimpMyStremio that handled being a server for an addon cluster that could run on LAN, to house locally running addons, (i think it even had a jackett addon) haven't heard anything about it for some time though, but it supported LAN SSL certificates as far as i can remember
there were community discussions about a possible revival of the project, not sure if that will happen or not though, but it may be worth looking into

**Comment by nyakaspeter** [comment]
Well I quickly realized that even with security concerns aside, localtunnel is not very reliable when their servers are overloaded. 

Thanks for the suggestion, I already knew PMS but I didn't know it handled SSL on LAN. Unfortunately it looks like it's not really maintained anymore. But I looked through it's documentation and the source code, it looks like it uses an external API to request a certificate for a domain derived from the server's LAN IP. I guess the domain is then pointing to the LAN IP. This approach can work if DNS rebinding protection doesn't block resolving domains to local addresses.

I found that [local-ip.co](http://local-ip.co/) offers a free service like this. It works better than localtunnel because the actual traffic to the addon does not have to leave the local network. Unfortunately it's still a third-party service that can go down anytime, but until then I might just go with it. Of course I could always set up my own reverse proxy with my own domain name and certificate, but I think that's too much to ask from others if they want to self-host the addon for themselves.

**Comment by besoeasy** [comment]
At this stage, it seems illogical that add-ons are not permitted to be accessed from a local area network (LAN).

@jaruba, I encourage you to consider the environmental impact of this limitation. For instance, I have an add-on running on my Raspberry Pi within my home network, but I am unable to use it on my Google TV without enforcing HTTPS. This forces me to route traffic through Cloudflare, ultimately resulting in unnecessary carbon emissions.

**Comment by chrisbog94** [comment]
I completely understand and share the frustration expressed in this issue... it's been open for over two years now.

For privacy-conscious users running personal Usenet/streaming addons (especially those with private indexers and NNTP credentials), the current requirement to either:

-> run everything on localhost, or
-> expose the service publicly with a real TLS certificate

which is a significant privacy regression. Forcing LAN-only services onto the public internet just to satisfy Stremio’s strict manifest fetcher defeats the entire purpose of self-hosting.

A simple checkbox in the addon installation dialog ("Allow insecure (HTTP) manifest URLs on local networks") would solve this for advanced users who are already running Docker, reverse proxies, and custom networks. We’re capable of accepting the risk ourselves.

Unfortunately, after spending several hours fighting self-signed certs, hosts-file hacks, and root-CA imports across multiple devices,, I’ve given up. I’ll just continue to download and store my media locally.

Thanks for keeping the issue alive. Perhaps I'll revisit this project in the future - it has potential.

Never-the-less, I'm disappointed.

**Comment by Raicuparta** [comment]
In case anyone else comes accross this, I'll share my findings. I used to be able to set local IP addons via http in the Android app, but that seems to have stopped working for some reason. I've found that you can bypass the https verification using Stremio's internal API This endpoint `https://api9.strem.io/api/addonCollectionSet` lets you change your addon list, and it seems like it skips verification altogether. You can look up what the actual shape of the POST request needs to be, and how to get the credentials (or ask your favorite LLM).

**Comment by nyakaspeter** [comment]
@Raicuparta thanks, this is actually a genius idea, I've implemented it in my addon: https://github.com/torrentstream/torrent-stream/blob/main/src/components/install-dialog.tsx

**Comment by anebot** [comment]
I’m running into the same issue.

The service is behind Caddy, configured as a reverse proxy with TLS enabled. stremio.local is reachable, and the SSL certificate appears to be valid. However, I still do not see the “**Add Addon**” button.

Is there any additional configuration required, or am I missing something?

**Comment by Raicuparta** [comment]
I made a simple node script to help me with this:
https://github.com/Raicuparta/stremio-addon-editor

**Comment by blaudp79** [comment]
> I made a simple node script to help me with this: https://github.com/Raicuparta/stremio-addon-editor

This works very well. Thank you!

**Comment by proxiangel** [comment]
The java script works great, but functionality does not. I changed to http://localhost and the addon I need works OK (I make an ssh tunnel to trick it that it's on localhost 80)
When I change to http://192.168.10.151 (where my service actually works) , addon does not work. 
I used the script to move the addon on top of the list, that didn't help.
Any ideas?

And honestly, security is great, but as stremio accepts localhost, I think it will not harm security if it also accepts all private networks (10.0.0.0/8 etc...)
